js/popup.js

```diff
@@ -1,16 +1,19 @@
-updateAllButtons();
+let allChanges = {};
+let isRunning = false;
 chrome.storage.onChanged.addListener(function (changes, namespace) {
     for (let [key, {
         oldValue,
         newValue
     }] of Object.entries(changes)) {
-        if (newValue != null) { // when save data changes, re-parse clean json with new save data and update buttons
+        Object.keys(changes).forEach((change) => allChanges[change] = true);
+        if (Object.keys(allChanges).length === 3 && !isRunning) { // when save data changes, re-parse clean json with new save data and update buttons
```

Comment:
I don't 100% understand this logic, but it changes the behavior of the extension in a way that is undesired. 

Now whenever somebody switches tabs, Idleon needs to be re-loaded. I don't like this behavior because people will often switch tabs to copy data back and forth. This needs to be changed so when popup is ran, it checks for existing save data and shows the copy buttons if it exists. The save data is then removed when the idleon window is launched or reloaded (which still happens).

Does this make sense?